grid-proxy/internal/indexer/indexer.go

```diff
@@ -0,0 +1,54 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/peer"
+)
+
+const (
+	indexerCallTimeout = 10 * time.Second
+)
+
+type Watcher interface {
+	Start(ctx context.Context)
+}
+
+type Indexer struct {
+	Watchers  map[string]Watcher
+	Paused    bool
+	Context   context.Context
+	RmbClient *peer.RpcClient
+}
+
+func NewIndexer(
+	ctx context.Context,
+	paused bool,
```

Comment:
Shouldn't be here, if for internal state, it should be defaulted to the zero value, false